dvc/repo/plots/__init__.py

```diff
@@ -459,6 +465,28 @@ def _resolve_definitions(
     return result
 
 
+def _add_annotations_to_image_definition(target):
+    if "data" not in target:
+        return target
+
+    path_to_remove = []
+    for path in target["data"]:
+        annotation_file = Path(path).with_suffix(".json").as_posix()
+        if ImageRenderer.matches(path) and annotation_file in target["data"]:
+            annotations = {"annotations": annotation_file}
+            # empty dict all share the same reference, so override them
+            if target["data"][path]:
+                target["data"][path].update({"annotations": annotation_file})
+            else:
+                target["data"][path] = annotations
+            path_to_remove.append(annotation_file)
```

Comment:
Can you explain a bit what's happening here?

Reply:
Let me know if it is clear now ;)